src/components/MagicCodeInput.tsx

```diff
@@ -1,98 +1,81 @@
     /** Name attribute for the input */
-    name: PropTypes.string,
+    name?: string;
 
     /** Input value */
-    value: PropTypes.string,
+    value?: string;
 
     /** Should the input auto focus */
-    autoFocus: PropTypes.bool,
+    autoFocus?: boolean;
 
     /** Whether we should wait before focusing the TextInput, useful when using transitions  */
-    shouldDelayFocus: PropTypes.bool,
+    shouldDelayFocus?: boolean;
 
     /** Error text to display */
-    errorText: PropTypes.string,
+    errorText?: string;
 
     /** Specifies autocomplete hints for the system, so it can provide autofill */
-    autoComplete: PropTypes.oneOf(['sms-otp', 'one-time-code', 'off']).isRequired,
+    autoComplete: 'sms-otp' | 'one-time-code' | 'off';
 
     /* Should submit when the input is complete */
-    shouldSubmitOnComplete: PropTypes.bool,
-
-    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+    shouldSubmitOnComplete?: boolean;
 
     /** Function to call when the input is changed  */
-    onChangeText: PropTypes.func,
+    onChangeText?: (value: string) => void;
 
     /** Function to call when the input is submitted or fully complete */
-    onFulfill: PropTypes.func,
+    onFulfill?: (value: string) => void;
 
     /** Specifies if the input has a validation error */
-    hasError: PropTypes.bool,
+    hasError?: boolean;
 
     /** Specifies the max length of the input */
-    maxLength: PropTypes.number,
+    maxLength?: number;
 
     /** Specifies if the keyboard should be disabled */
-    isDisableKeyboard: PropTypes.bool,
+    isDisableKeyboard?: boolean;
 
     /** Last pressed digit on BigDigitPad */
-    lastPressedDigit: PropTypes.string,
+    lastPressedDigit?: string;
 
     /** TestID for test */
-    testID: PropTypes.string,
+    testID?: string;
 };
 
-const defaultProps = {
-    value: '',
-    name: '',
-    autoFocus: true,
-    shouldDelayFocus: false,
-    errorText: '',
-    shouldSubmitOnComplete: true,
-    innerRef: null,
-    onChangeText: () => {},
-    onFulfill: () => {},
-    hasError: false,
-    maxLength: CONST.MAGIC_CODE_LENGTH,
-    isDisableKeyboard: false,
-    lastPressedDigit: '',
-    testID: '',
+type MagicCodeInputHandle = {
+    focus: () => void;
+    focusLastSelected: () => void;
+    resetFocus: () => void;
+    clear: () => void;
+    blur: () => void;
 };
 
 /**
  * Converts a given string into an array of numbers that must have the same
  * number of elements as the number of inputs.
- *
- * @param {String} value
- * @param {Number} length
- * @returns {Array}
  */
-const decomposeString = (value, length) => {
-    let arr = _.map(value.split('').slice(0, length), (v) => (ValidationUtils.isNumeric(v) ? v : CONST.MAGIC_CODE_EMPTY_CHAR));
+const decomposeString = (value: string, length: number): string[] => {
+    let arr = value
+        .split('')
+        .slice(0, length)
+        .map((v) => (ValidationUtils.isNumeric(v) ? v : CONST.MAGIC_CODE_EMPTY_CHAR));
```

Comment:
@mountiny I am not sure if we add types to the params in array functions here? Can you confirm?

Reply:
asking

Reply:
@mananjadhav No need, these params should be inferred 👍